torch/onnx/_internal/exporter.py

```diff
@@ -419,34 +578,35 @@ def __init__(
         model_args: Sequence[Any],
         model_kwargs: Mapping[str, Any],
     ):
-        if isinstance(options, ExportOptions):
-            self.options = ResolvedExportOptions(options)
-        elif isinstance(options, ResolvedExportOptions):
+        if isinstance(options, ResolvedExportOptions):
             self.options = options
+        elif isinstance(options, ExportOptions):
+            self.options = ResolvedExportOptions(options)
```

Comment:
Maybe just make `ResolvedExportOptions()` idempotent to combine both branches?

Reply:
Not sure what you meant

Reply:
Instead of casing here, move this logic to the initializer of `ResolvedExportOptions` so we can collapse the isinstance calls on the call site to simply

```
self.options = ResolvedExportOptions(options)
```

always.

Reply:
Oh, I did that the first time, but then I changed because:
* beartype didn't like `ResolvedExportOptions.__init__(self, options: Optional[Union[ExportOptions, ResolvedExportOptions]])`, so I had to change it to `Optional[Union[ExportOptions, "ResolvedExportOptions"]]` to prevent `beartype.roar.BeartypePep563Exception: @beartyped torch.onnx._internal.exporter.ResolvedExportOptions.__init__() parameter "options" PEP 563-postponed type hint 'Optional[Union[ExportOptions, ResolvedExportOptions]]' syntactically invalid (i.e., "name 'ResolvedExportOptions' is not defined") `
* Python doesn't have copy constructor, so I would have to manually assign all `options`'s properties to `self`, making yet another redundancy assignment. Everytime we add a new option, we would have to change in "fake copy ctor".

I wonder why you didn't say anything when you approved https://github.com/pytorch/pytorch/pull/97920/files#diff-0795f54fd1f38cfbf2c4a863a4efc9f40f2ea020a2b1612605c361b8d8d35862R218-R222 which introduced this branching on `Exporter` instead of `ResolvedExportOptions`. My only change was to switch the order of comparison, but the branching was there already :) 

Nonetheless, I did this change as suggested.